lib/sepa_reason_codes/code.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module SepaReasonCodes
+  class Code
+    ATTRS = %i[code iso_name description probable_status].freeze
+
+    attr_reader(*ATTRS)
+
+    def initialize(code:, iso_name:, description:, probable_status:)
+      @code = code
+      @iso_name = iso_name
+      @description = description
+      @probable_status = probable_status
+    end
+
+    def ==(other)
+      ATTRS.all? do |attribute|
```

Comment:
I don't know a lot about the underlying data, but my initial impression is that solely the `code` needs to be comparable to determine equality, not all attributes.

Reply:
In the beginning, I went for checking just for code, but didn't like a partial check on the instance. 

So decided to check for all values, as it should be same.